fog/ledger/server/src/router_server.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+use futures::executor::block_on;
+use mc_common::logger::{log, Logger};
+use mc_fog_api::ledger_grpc;
+use mc_fog_ledger_enclave::LedgerEnclaveProxy;
+use mc_fog_uri::{ConnectionUri, KeyImageStoreUri};
+use mc_util_grpc::{ConnectionUriGrpcioServer, ReadinessIndicator};
+
+use crate::{
+    config::LedgerRouterConfig, router_admin_service::LedgerRouterAdminService,
+    router_service::LedgerRouterService,
+};
+
+pub struct LedgerRouterServer {
+    router_server: grpcio::Server,
+    admin_server: grpcio::Server,
+    logger: Logger,
+}
+
+impl LedgerRouterServer {
+    pub fn new<E>(
+        config: LedgerRouterConfig,
+        enclave: E,
+        shards: Arc<RwLock<HashMap<KeyImageStoreUri, Arc<ledger_grpc::KeyImageStoreApiClient>>>>,
+        logger: Logger,
+    ) -> LedgerRouterServer
+    where
+        E: LedgerEnclaveProxy,
+    {
+        let readiness_indicator = ReadinessIndicator::default();
+
+        let env = Arc::new(
+            grpcio::EnvBuilder::new()
+                .name_prefix("key-image-router-server".to_string())
+                .build(),
+        );
+
+        // Health check service - will be used in both cases
+        let health_service =
+            mc_util_grpc::HealthService::new(Some(readiness_indicator.into()), logger.clone())
+                .into_service();
+
+        // Build our router server.
+        // Init ledger router service.
+        let ledger_router_service = ledger_grpc::create_ledger_api(LedgerRouterService::new(
+            enclave,
+            shards.clone(),
+            config.query_retries,
+            logger.clone(),
+        ));
+        log::debug!(logger, "Constructed Ledger Router GRPC Service");
+
+        // Init ledger router admin service.
+        let ledger_router_admin_service = ledger_grpc::create_ledger_router_admin_api(
+            LedgerRouterAdminService::new(shards, logger.clone()),
+        );
+        log::debug!(logger, "Constructed Ledger Router Admin GRPC Service");
+
+        // Package service into grpc server
+        log::info!(
+            logger,
+            "Starting Key Image Router server on {}",
```

Comment:
:upside_down_face: Similar to above should this say ledger instead of key image?
```suggestion
            "Starting Ledger Router server on {}",
```